### ISSUE TITLE
Crash in TaskDelegate 

### ISSUE THREAD

**Issue by LaurentLK** [Alamofire/Alamofire#1708]
Randomly have crashes in TaskDelegate from download task on background session on line 335: 

```
        let result = destination(location, downloadTask.response as! HTTPURLResponse)
```

The download task response is nil that's why it's crashing. Is it normal that there is no safe unwrapping ?

**Comment by jshier** [comment]
Thanks for the report, this definitely looks incorrect. We'll try to get a fix out this weekend.

**Comment by jshier** [comment]
This will be fixed when #1723 lands.
